### PR TITLE
move checking for ledger version 7 to SignatureChecker

### PR DESCRIPTION
**Pull request by vogel**
Signed-off-by: Rafał Malinowski <rafal.przemyslaw.malinowski@gmail.com>

### CODE REVIEW

File: src/transactions/SignatureChecker.cpp [hunk 1]
```diff
@@ -18,9 +18,10 @@ using xdr::operator<;
 using xdr::operator==;
 
 SignatureChecker::SignatureChecker(
+    bool alwaysOk,
     Hash const& contentsHash,
     xdr::xvector<DecoratedSignature, 20> const& signatures)
-    : mContentsHash(contentsHash), mSignatures(signatures)
+    : mAlwaysOk{alwaysOk}, mContentsHash{contentsHash}, mSignatures{signatures}
```
Comment (MonsieurNicolas) [inline src/transactions/SignatureChecker.cpp]: I don't think adding a bool like this is the right approach: if we make any changes to the way we check signatures, we'll need the protocol version, so I'd rather have the protocol version passed in.

**Comment by MonsieurNicolas** [comment]
Change looks good now, thank you

**Comment by vogel** [comment]
Ready for merge

**Comment by MonsieurNicolas** [comment]
r+ 24d15ffcfa3de27acf8cefa3706379155ef4741d

**Pull request merged**